_posts/2019-08-04-plonk-stack.md

```diff
@@ -0,0 +1,269 @@
+
+![](https://docs.openfaas.com/images/of-conceptual-operator.png)
+
+*[Architecture diagram](https://docs.openfaas.com/architecture/gateway/) showing OpenFaaS with CRDs*
+
+Today OpenFaaS runs on Kubernetes much the same as it did in 2017, with a few enhancements:
+
+* HTTP probes are the default for efficient scaling
+* Scale to and from zero was added
+* CRDs and an operator are available
+* A helm chart was developed and is now the community's favourite way to get OpenFaaS
+* [Stateless microservices](https://docs.openfaas.com/reference/workloads/) became a first-class citizen along with the existing functions
+* Support for secrets
+* The "of-watchdog" was developed to enable either STDIO or HTTP to be the interface to your code
+
+You can get a good overview of what the community has been building from the 2019 Project Update including some of the customer journeys from KubeCon.
+
+> See also: [OpenFaaS 2019 Update](https://www.openfaas.com/blog/project-update/)
+
+Everything deployed with OpenFaaS uses an idiomatic approach, so that things are exactly where you would expect them to be, without magic. For that reason you can use your favourite commands from the [kubectl cheatsheet](https://kubernetes.io/docs/reference/kubectl/cheatsheet/).
+
+> You can run the whole PLONK stack along with a light-weight distribution of Kubernetes such as [k3s from Rancher](https://k3s.io) in as little a 1GB RAM. That leaves a lot of headroom for what you really care about. This makes the stack well suited to IoT, Edge, and datacentre workloads.
+
+You can [get Kubernetes](https://kubernetes.io/) from your favourite cloud as a service, in your own datacentre, or run it on your laptop.
+
+#### Summing up PLONK
+
+* FaaS or PaaS?
+
+    In the introduction I mentioned that OpenFaaS can be used as a FaaS or a PaaS. I believe that "FaaS" really is only a specialization of PaaS adding in a templating system for creating and managing code. Given how OpenFaaS has changed since it was originally developed, it makes a feature-rich platform for deploying any kind of services.
+
+    <blockquote class="twitter-tweet" data-conversation="none"><p lang="en" dir="ltr">What does it get you? <a href="https://t.co/GrQCrYtKNr">pic.twitter.com/GrQCrYtKNr</a></p>&mdash; Alex Ellis (@alexellisuk) <a href="https://twitter.com/alexellisuk/status/1154385354755125248?ref_src=twsrc%5Etfw">July 25, 2019</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+* What comes after OpenFaaS?
+
+    In late 2017 I started to design a distribution of OpenFaaS that shipped as a whole package including HTTPS, multi-user support, authz and CI/CD with GitHub and GitLab.
+
+    It's called OpenFaaS Cloud and is free and open-source. You can host your own using the [ofc-bootstrap tool](https://github.com/openfaas-incubator/ofc-bootstrap) or [request free access to the Community Cluster](https://github.com/openfaas/community-cluster/).
```

Comment:
OpenFaaS already builds on top of PLONK stack.

Reply:
OpenFaaS Cloud doesn't ship with Linkerd today, but I think we could package it for future users as an optional component. cc @matipan